cmd/generate-catalog/generate_test.go

```diff
@@ -416,206 +351,78 @@ func TestGenerateEmptyChannels(t *testing.T) {
-			versions: []*semver.Version{
-				semver.MustParse("4.0.0"),
-			},
-			channels: []Channel{
-				{Name: "latest"},
-			},
-			oldestSupportedVersion: semver.MustParse("4.0.0"),
-			validate: func(t *testing.T, deprecations Deprecations) {
-				hasLatestDeprecation := false
-				for _, entry := range deprecations.Entries {
-					if entry.Reference.Schema == olmChannelSchema && entry.Reference.Name == "latest" {
-						hasLatestDeprecation = true
-					}
-				}
-				assert.True(t, hasLatestDeprecation, "Latest channel should be deprecated")
-			},
-		},
-		{
-			name: "No deprecations when all versions are supported",
-			versions: []*semver.Version{
-				semver.MustParse("4.0.0"),
-				semver.MustParse("4.1.0"),
-			},
-			channels: []Channel{
-				{Name: "rhacs-4.0", yStreamVersion: semver.MustParse("4.0.0")},
-				{Name: "rhacs-4.1", yStreamVersion: semver.MustParse("4.1.0")},
-			},
-			oldestSupportedVersion: semver.MustParse("4.0.0"),
-			validate: func(t *testing.T, deprecations Deprecations) {
-				assert.Equal(t, 1, len(deprecations.Entries))
-				assert.Equal(t, olmChannelSchema, deprecations.Entries[0].Reference.Schema)
-				assert.Equal(t, "latest", deprecations.Entries[0].Reference.Name)
-			},
-		},
+	versions := []*semver.Version{
+		semver.MustParse("3.62.0"),
+		semver.MustParse("3.62.1"),
+		semver.MustParse("4.0.0"),
+		semver.MustParse("4.1.0"),
 	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			deprecations := generateDeprecations(tt.versions, tt.channels, tt.oldestSupportedVersion)
-
-			assert.Equal(t, olmDeprecationsSchema, deprecations.Schema)
-			assert.Equal(t, rhacsOperator, deprecations.Package)
-			if tt.validate != nil {
-				tt.validate(t, deprecations)
-			}
-		})
+	channels := []Channel{
+		{Name: "rhacs-3.62", yStreamVersion: semver.MustParse("3.62.0")},
+		{Name: "latest"},
+		{Name: "rhacs-4.0", yStreamVersion: semver.MustParse("4.0.0")},
+		{Name: "rhacs-4.1", yStreamVersion: semver.MustParse("4.1.0")},
+		{Name: "stable"},
 	}
+	oldestSupportedVersion := semver.MustParse("4.0.0")
+
+	deprecations := generateDeprecations(versions, channels, oldestSupportedVersion)
+
+	assert.Equal(t, olmDeprecationsSchema, deprecations.Schema)
+	assert.Equal(t, rhacsOperator, deprecations.Package)
+
+	// Should deprecate:
+	// - `latest` channel
+	// - `rhacs-3.62` channel
+	// - `rhacs-operator.v3.62.0` bundle
+	// - `rhacs-operator.v3.62.1` bundle
+	assert.Len(t, deprecations.Entries, 4)
+	assertDeprecationEntry(t, deprecations.Entries[0], olmChannelSchema, "latest", latestChannelDeprecationMessage)
+	assertDeprecationEntry(t, deprecations.Entries[1], olmChannelSchema, "rhacs-3.62", channelDeprecationMessage)
+	assertDeprecationEntry(t, deprecations.Entries[2], olmBundleSchema, "rhacs-operator.v3.62.0", bundleDeprecationMessage)
+	assertDeprecationEntry(t, deprecations.Entries[3], olmBundleSchema, "rhacs-operator.v3.62.1", bundleDeprecationMessage)
```

Comment:
One improvement suggestion is that this test currently depends on the particular order of entries. This is actually not important for OLM and for end users, as well the order isn't that important for us. What is important that necessary entries are included (no matter in which position of the slice).

Since you already use a helper function, it could be slightly modified to allow for such use:
```suggestion
	assertContainsDeprecationEntry(t, deprecations.Entries, olmChannelSchema, "latest", latestChannelDeprecationMessage)
	assertContainsDeprecationEntry(t, deprecations.Entries, olmChannelSchema, "rhacs-3.62", channelDeprecationMessage)
	assertContainsDeprecationEntry(t, deprecations.Entries, olmBundleSchema, "rhacs-operator.v3.62.0", bundleDeprecationMessage)
	assertContainsDeprecationEntry(t, deprecations.Entries, olmBundleSchema, "rhacs-operator.v3.62.1", bundleDeprecationMessage)
```

This isn't very important in the current state so up to you if you want to implement.